### ISSUE TITLE
Add python APIs for serialization that accept modules as string arguments

### ISSUE THREAD

**Issue by GleasonK** [openxla/stablehlo#1361]
### Request description

Users that have StableHLO built in a separate shared library cannot use the current API's since passing an MLIRContext across shared library boundaries causes issues.

```
def serialize_portable_artifact_str(module_str, target) -> bytes
def deserialize_portable_artifact_str(artifact_bytes):  -> str
```

**Comment by burmako** [comment]
Should we perhaps just change the existing API to take a string instead of a module?

**Comment by GleasonK** [comment]
Closed by #1461
